pkg/kubelet/dockershim/docker_container.go

```diff
@@ -383,9 +383,23 @@ func (ds *dockerService) ContainerStatus(containerID string) (*runtimeApi.Contai
 	}, nil
 }
 
-// Exec execute a command in the container.
-// TODO: Need to handle terminal resizing before implementing this function.
-// https://github.com/kubernetes/kubernetes/issues/29579.
-func (ds *dockerService) Exec(containerID string, cmd []string, tty bool, stdin io.Reader, stdout, stderr io.WriteCloser) error {
+// ExecLegacy execute a command in the container.
+// TODO: remove this after full CRI streaming APIs are accomplished.
+func (ds *dockerService) ExecLegacy(containerID string, cmd []string, tty bool, stdin io.Reader, stdout, stderr io.WriteCloser) error {
 	return fmt.Errorf("not implemented")
 }
+
+// Exec prepares a streaming endpoint to execute a command in the container.
```

Comment:
Are all these functions here and below are expected to be implemented in the future? If so we need add TODO in case of misunderstanding.